.github/workflows/tests.yml

```diff
@@ -0,0 +1,48 @@
+name: Tests
+
+on: [push, pull_request]
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: true
+            matrix:
+                php: [7.2, 7.3, 7.4, 8.0, 8.1]
+                stability: [prefer-lowest, prefer-stable]
+
+        name: PHP ${{ matrix.php }} - ${{ matrix.stability }}
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php }}
+                  extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, bcmath, soap, intl, gd, exif, iconv, imagick, fileinfo
+                  coverage: none
+
+            - name: Setup problem matchers
+              run: |
+                  echo "::add-matcher::${{ runner.tool_cache }}/php.json"
+                  echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+            - name: Validate composer.json
+              run: composer validate --strict
+
+            - name: Lint files
+              run: find src tests -name '*.php' | xargs -n 1 -P4 php -l
+
+            - name: Install dependencies
+              run: composer update --${{ matrix.stability }} --prefer-dist --no-interaction
+
+            - name: Execute tests
+              run: vendor/bin/phpunit --fail-on-risky
```

Comment:
Just a question. Will phpunit be ran for all versions of php with the preferred stability?

Reply:
Yes, it uses the dependencies that were installed in the previous step